src/main/java/pl/isa/biblioteka/Menu.java

```diff
@@ -9,34 +9,42 @@ public void selectUser() {
         System.out.println("1. Bibliotekarz");
         System.out.println("2. Użytkownik");
         Scanner scanner = new Scanner(System.in);
-        try {
-            int userChoose = scanner.nextInt();
-            if (userChoose == 1) {
-                librarianMenu();
-            } else if (userChoose == 2) {
-                userMenu();
+        boolean isContinue = true;
+        while (isContinue) {
```

Comment:
Mając pętle nie potrzebujesz rekurencji, czyli kolejnego wywołania metody selectUser. Jak usuniesz te wywołania to zobaczysz, że nie działa Ci kolejne wczytanie, bo teraz kolejne wywołanie funkcji tworzy nowego Scannera, a przy nextInt jak chcemy coś kolejnego wczytać, to trzeba wczytać znak nowej linii, który zostaje w strumieniu wejściowym po kliknięciu entera. Do przerobienia ta pętla :)

Reply:
Było dokładnie jak piszesz nie działał program poprawnie dlatego wrzuciłem powrót do menu i rozwiązało problem.
Ok przebudowałem menu pozbyłem się zbędnych if, ale pojawił się problem z zamykaniem aplikacji czy mogę w takiej sytuacji użyć metody System.Exit() . Inspect code nie zwraca żadnych warnings kod działa jak tego oczekuję, ale nie wiem czy tak jest ok.

Reply:
Jest ok ;)

Reply:
Aczkolwiek u mnie `case 3 -> isContinue = false;` działa... :)

Reply:
Działa za 1 razem :D 
Jak wejdziesz do menu np usera czy bibliotekarza i wrócisz opcją zmiany usera to już go nie zamyka :D 
Zabawa jak z tym memem od CSS i i żaluzjami ;)
A na poważnie dzięki za wskazówki myślę, że ta wersja jest czytelniejsza i bez zbędnych if ....